### PR TITLE
Added Invite Placement Coordinator functionality

### PR DESCRIPTION
**Pull request by harshalijain**
Added Invite Placement Coordinator functionality

### CODE REVIEW

File: www/components/common/utils/constants.js [hunk 1]
```diff
@@ -22,7 +22,10 @@ var HttpRoutes = {
     //Coordinator specific routes
     Coordinator : {
         personalInfo: apiUrl + '/coordinators/{id}/user',
-        collegeDetail: apiUrl + '/colleges/{id}/creator'
+        collegeDetail: apiUrl + '/colleges/{id}/creator',
+        invitedPlacementCoordinators: apiUrl + '/colleges/{college_id}/coordinators',
```
Comment (piyushbeli) [inline www/components/common/utils/constants.js]: it should be placeCoordinators not invitePlaceCoordinators, becaue it fetches the list of coordinators.

File: www/app.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 angular.module('app', ['ionic', 'ng-token-auth', 'ui.router', 'app.login', 'app.verifyMe',
-    'app.user.detail', 'app.college.detail', 'app.config', 'app.menu', 'app.cache', 'app.placement.activities',
+    'app.user.detail', 'app.college.detail', 'app.config', 'app.menu', 'app.cache', 'app.placement.activities','app.invite.PlacementCoordinator',
```
Comment (piyushbeli) [inline www/app.js]: Follow camel case when you define a module name.
Correct:  app.invite.placementCoordinator

File: www/components/coordinator/invitePlacementCoordinator/invite.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+appInvitePlacementCoordinator.factory("Invite", function() {
+
+	function Invite(data) {
+		if(!data){
+            return;
```
Comment (piyushbeli) [inline www/components/coordinator/invitePlacementCoordinator/invite.js]: Correct the indentation.

File: www/components/coordinator/invitePlacementCoordinator/invite.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        }
+
+        this.id = data.id;
+        this.email = data.email;
+        this.joined = data.joined;
+
```
Comment (piyushbeli) [inline www/components/coordinator/invitePlacementCoordinator/invite.js]: Remove empty line.

File: www/components/coordinator/invitePlacementCoordinator/invitePlacementCoordinator.html [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+       <label class="item-input-wrapper">
+          <input type="text" ng-model="inviteEmail" placeholder="Email address">
+       </label>
+       <button class="button button-small" ng-click="invitePlacementCoordinator(inviteEmail)">
+          Invite
+       </button>
+     </div>
+     <ion-list ng-repeat="invite in invites">
+      <ion-item>  
+
```
Comment (piyushbeli) [inline www/components/coordinator/invitePlacementCoordinator/invitePlacementCoordinator.html]: Empty line

File: www/components/coordinator/invitePlacementCoordinator/invitePlacementCoordinatorController.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
+var appInvitePlacementCoordinator = angular.module("app.invite.PlacementCoordinator", []);
+
+appInvitePlacementCoordinator.controller("InvitePlacementCoordinatorController",function($scope,$ionicModal,UserService,
+	InvitePlacementCoordinatorService) {
+
+	UserService.getCurrentUser()
+          .then(function(user) {
+                $scope.currentUser = user;
```
Comment (piyushbeli) [inline www/components/coordinator/invitePlacementCoordinator/invitePlacementCoordinatorController.js]: Although it diesn't matter a lot, but in other module we have used $scope.user not currentUser, so you may want to change here also.

File: www/components/coordinator/invitePlacementCoordinator/invitePlacementCoordinatorController.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+                   })
+                   .catch(function (errorMessage) {
+                        alert(errorMessage);
+                });
+        };
+  
+        $scope.removePlacementCoordinator = function(inviteId) {
+                InvitePlacementCoordinatorService.removePlacementCoordinator(inviteId)
+                   .then(function(success) {
+                        for(var i = $scope.invites.length - 1; i >= 0; i--){
```
Comment (piyushbeli) [inline www/components/coordinator/invitePlacementCoordinator/invitePlacementCoordinatorController.js]: There should not be any login in the controller, move this loop into the service and return the invites from there and assign into the scope.

File: www/components/coordinator/invitePlacementCoordinator/invitePlacementCoordinatorService.js [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+   	  var deferred = $q.defer(),
+   	  	  url = self.invitedPlacementCoordinatorsUrl().format({college_id: user.collegeId});
+
+   	  $http.get(url)
+   	    .then(function(response) {
+   	    	deferred.resolve(Invite.build(response.data));
+   	    })
+   	    .catch(function (error) {
+                var errorMessage = error.message || "Error occurred while fetching Invited Coordinator detail ";
```
Comment (piyushbeli) [inline www/components/coordinator/invitePlacementCoordinator/invitePlacementCoordinatorService.js]: Change the error message as per our api.

File: www/components/coordinator/invitePlacementCoordinator/invitePlacementCoordinatorService.js [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+
+   self.invitePlacementCoordinator = function (inviteEmail,user) {
+   	   var deferred = $q.defer(),
+   	       url = self.invitePlacementCoordinatorUrl() ,
+   	       postData = 
+   	       { 
+   	           college_id: user.collegeId,
+   	           email: inviteEmail 
+   	       },
+   	       requestType = HttpRequestType.GET;
```
Comment (piyushbeli) [inline www/components/coordinator/invitePlacementCoordinator/invitePlacementCoordinatorService.js]: No need to keep it in a variable. you can directly assign it in req

File: www/components/coordinator/invitePlacementCoordinator/invitePlacementCoordinatorService.js [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+   	       	    var newInviteData = 
+   	            {
+   	        	   id: response.data.id,
+   	        	   email: inviteEmail,
+   	        	   joined: false
+   	            };
+   	       	    deferred.resolve(Invite.build(newInviteData)); 
+   	       })
+   	        .catch(function (error) {
+
```
Comment (piyushbeli) [inline www/components/coordinator/invitePlacementCoordinator/invitePlacementCoordinatorService.js]: remove empty line

File: www/components/coordinator/invitePlacementCoordinator/invitePlacementCoordinatorService.js [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+   self.removePlacementCoordinator = function(inviteId) {
+   	    var deferred = $q.defer(),
+   	    url = self.removePlacementCoordinatorUrl().format({ inviteId : inviteId }),
+   	    requestType = HttpRequestType.DELETE;
+   	    
+   	    var req = 
+   	    {
+   	    	url: url,
+   	    	method : requestType
```
Comment (piyushbeli) [inline www/components/coordinator/invitePlacementCoordinator/invitePlacementCoordinatorService.js]: same as above.

File: www/components/coordinator/invitePlacementCoordinator/invitePlacementCoordinatorService.js (hunk 8 above)
Comment (harshalijain) [inline www/components/coordinator/invitePlacementCoordinator/invitePlacementCoordinatorService.js]: Hi,
On this suggestion, what exactly is to be displayed as error message. Do I
need to show parameters in error message for which the api was hit? I am
unclear about this one.

-Harshali Jain
   MetaCube Software Pvt. Ltd.
   Jaipur, Rajasthan (India)

On Fri, Feb 27, 2015 at 5:36 PM, piyushbeli <notifications@github.com>
wrote:

> In
> www/components/coordinator/invitePlacementCoordinator/invitePlacementCoordinatorService.js
> <https://github.com/piyushbeli/metacampus-placement/pull/1#discussion_r25502267>
> :
>
> > +
> > +   self.removePlacementCoordinatorUrl = function() {
> > +   	  return HttpRoutes.Coordinator.removeCoordinators;
> > +   };
> > +
> > +   self.fetchInvitedPlacementCoordinators = function(user) {
> > +
> > +   	  var deferred = $q.defer(),
> > +   	  	  url = self.invitedPlacementCoordinatorsUrl().format({college_id: user.collegeId});
> > +
> > +   	  $http.get(url)
> > +   	    .then(function(response) {
> > +   	    	deferred.resolve(Invite.build(response.data));
> > +   	    })
> > +   	    .catch(function (error) {
> > +                var errorMessage = error.message || "Error occurred while fetching Invited Coordinator detail ";
>
> Change the error message as per our api.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/piyushbeli/metacampus-placement/pull/1/files#r25502267>
> .
>

______________________________________________________________________________________________________
NOTICE: This email and any files transmitted with it, including their content, are confidential and intended solely for the use of the individual or entity to whom they are addressed. If you have received this email by mistake, or in error, please notify the sender and/or system manager immediately and delete this e-mail from your system. If you are not the named or intended addressee of this email then you should not disseminate, distribute, copy or disclose this e-mail and you are strictly prohibited from taking any action in reliance on the information contained in this email and any files attached with it.


Although the company has taken reasonable precautions to ensure no viruses are present in this email, the company cannot accept responsibility for any loss or damage arising from the use of this email or attachments.
______________________________________________________________________________________________________

Comment (piyushbeli) [inline www/components/coordinator/invitePlacementCoordinator/invitePlacementCoordinatorService.js]: "Error occurred while fetcing invited coordinators" We are not fetching detail.

**Pull request merged**